app/main.py

```diff
@@ -1,34 +1,86 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.colum = column
+        self.is_alive = is_alive
+        self.coordinates = (self.row, self.colum)
+
+    def __repr__(self) -> str:
+        return f"{self.coordinates}"
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+        self.start = Deck(start[0], start[1])
+        self.end = Deck(end[0], end[1])
+        self.is_drowned = is_drowned
+        self.decks = [self.start]
+        if self.start.coordinates != self.end.coordinates:
+            self.create_list_of_decks()
+
+    def __repr__(self) -> str:
+        return f"{self.decks}"
+
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.colum == column:
+                return deck
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def fire(self, row: int, column: int) -> None:
+        deck = self.get_deck(row, column)
+        if deck.is_alive:
+            deck.is_alive = False
```

Comment:
No need if statement here
```suggestion
        deck.is_alive = False
```